e2e/app.e2e-spec.ts

```diff
@@ -23,4 +23,21 @@ describe('Tree App', () => {
     expect(firstAsyncChild.getText()).toEqual('Input Mono');
     expect(page.getLastAsyncChild().getText()).toEqual('Source Code Pro');
   });
+
+  it('Should render tree node with HTML tags', () => {
+    page.navigateTo();
+
+    const firstChildNode = page.getFirstChildNode();
+    expect(browser.isElementPresent(firstChildNode)).toBeTruthy();
+    expect(firstChildNode.getText()).toEqual('Antiqua');
+
+    // verify if the element attributes are same
+    const attrs = {'id': 'antiqua', 'class': 'test'};
+
+    [].forEach.call(Object.keys(attrs), (key) => {
+      firstChildNode.getAttribute(key).then((value) => {
```

Comment:
This is executed via promises, hence code is async so It is better to use Promise.all for checking this.

```typescript
const expectations = Object.keys(attrs).map((key) => {
    return firstChildNode.getAttribute(key).then((value) => expect(value).toEqual(attrs[key]));
});

// And then just wait and assert for
Promise.all(expectations)
```